### PR TITLE
Add the salt-overlay directory to the GCE master-pd and reserve the master's IP

### PR DESCRIPTION
**Pull request by a-robinson**
...upon creation to make it easier to replace the master later.

This pulls out the parts of PR #3174 that don't break anything and will make upgrading existing clusters in the future less painful. #3174 will still need more effort to make work, unfortunately.

### CODE REVIEW

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -454,28 +461,12 @@ function kube-up {
     echo "readonly DNS_SERVER_IP='${DNS_SERVER_IP:-}'"
     echo "readonly DNS_DOMAIN='${DNS_DOMAIN:-}'"
     grep -v "^#" "${KUBE_ROOT}/cluster/gce/templates/common.sh"
-    grep -v "^#" "${KUBE_ROOT}/cluster/gce/templates/format-and-mount-pd.sh"
+    grep -v "^#" "${KUBE_ROOT}/cluster/gce/templates/mount-pd.sh"
```
Comment (zmerlynn) [inline cluster/gce/util.sh]: Did you miss committing this file, or did I miss something?

Comment (a-robinson) [inline cluster/gce/util.sh]: Ugh, yeah. Thanks for catching that.

**Comment by a-robinson** [comment]
I've also added /etc/salt/, as /etc/salt/pki/ contains the dynamically generated RSA keys used by salt. Without saving it, the salt minions on the nodes don't trust a replacement master.

File: cluster/gce/templates/mount-pd.sh [hunk 2]
```diff
@@ -14,20 +14,34 @@
...
 mkdir -p /mnt/master-pd
 /usr/share/google/safe_format_and_mount -m "mkfs.ext4 -F" "${device_path}" /mnt/master-pd
+# Contains all the data stored in etcd
 mkdir -m 700 -p /mnt/master-pd/var/etcd
+# Contains the dynamically generated apiserver auth certs and keys
 mkdir -p /mnt/master-pd/srv/kubernetes
+# Contains the cluster's initial config parameters and auth tokens
+mkdir -p /mnt/master-pd/srv/salt-overlay
+# Contains salt's dynamically generated RSA keys, among other configuration
+mkdir -p /mnt/master-pd/etc/salt
```
Comment (zmerlynn) [inline cluster/gce/templates/mount-pd.sh]: If `/etc/salt/pki` is all you need, would it be safer just to handle that directory alone? (it's another `mkdir -m 700`, looks like).

I don't see anything immediately unsafe to mimick in the `/etc/salt` dir, but it feels it would be better to be surgical here.

File: cluster/gce/util.sh [hunk 3]
```diff
@@ -433,6 +433,13 @@ function kube-up {
     --target-tags "${MASTER_TAG}" \
     --allow tcp:443 &
 
+  # We have to make sure the disk is created before creating the master VM, so
+  # run this in the foreground.
+  gcloud compute disks create "${MASTER_NAME}-pd" \
+    --project "${PROJECT}" \
+    --zone "${ZONE}" \
+    --size "10GB"
+
```
Comment (zmerlynn) [inline cluster/gce/util.sh]: Can you explain the re-organizations in here, both of the `compute disks create` and the `MINION_SCOPES`? They don't seem to be necessary?

If it's just cleanup, maybe revert or put in a separate commit?

File: cluster/gce/templates/mount-pd.sh (hunk 2 above)
Comment (a-robinson) [inline cluster/gce/templates/mount-pd.sh]: Sure, done. Albeit with `-m 770`, since that's what I'm seeing on my cluster. Are you seeing 700?

File: cluster/gce/util.sh (hunk 3 above)
Comment (a-robinson) [inline cluster/gce/util.sh]: Yes, it's just cleanup to keep the commands grouped more logically. I'll move it into a separate commit. Is it fine if I leave it in this PR?

File: cluster/gce/templates/mount-pd.sh (hunk 2 above)
Comment (zmerlynn) [inline cluster/gce/templates/mount-pd.sh]: Sorry, you're right. It's 770.

File: cluster/gce/util.sh (hunk 3 above)
Comment (zmerlynn) [inline cluster/gce/util.sh]: Yeah, I'm happy in the same PR. I was just trying to make sure there wasn't something I was missing.

**Comment by zmerlynn** [comment]
LGTM. Will merge on Travis.

**Comment by zmerlynn** [comment]
Travis is being dumb, and won't show anything useful here. Rolling the dice.
